NetworkOpener.cpp

```diff
@@ -373,11 +372,11 @@ void NetworkOpener::publishAuthorizationInfo(const std::list<NukiOpener::LogEntr
         }
         --i;
 
-        if((log.loggingType == NukiOpener::LoggingType::LockAction || log.loggingType == NukiOpener::LoggingType::KeypadAction) && ! authFound)
+        if((log.loggingType == NukiOpener::LoggingType::LockAction || log.loggingType == NukiOpener::LoggingType::KeypadAction) && ! _authFound)
         {
-            authFound = true;
-            authId = log.authId;
-            memcpy(authName, log.name, sizeof(log.name));
+            _authFound = true;
+            _authId = log.authId;
+            _authName = (char*)log.name;
```

Comment:
This will be a problem when the log.name uses all 32 characters. In that case it will not be 0-terminated, and casting it to char will make it read the name until it randomly finds a 0 in the memory after log.name. You'll have to copy it into a temporary char[33] array before assigning it.

Reply:
e.g. like this (I'm trying to understand it)?
 
```
char tempName[33];  // temporary buffer
memset(tempName, 0, sizeof(tempName));  // initialize the buffer with zeros

// copy characters from log.name into temporary buffer
strncpy(tempName, log.name, sizeof(tempName) - 1);  // last place remains for the zero terminator

_authFound = true;
_authId = log.authId;
_authName = tempName;  // Assign _authName, secure with null terminator
```

@iranl, can we try that too, or do you have a better idea?

Reply:
I looked at https://arduinojson.org/v6/issues/garbage-out/ and guessed the problem could be most easily be solved by using a String. Might not be the best solution though. I've updated the PR, but lets see if the solution by @technyon works.